### ISSUE TITLE
[Bug]: paths creates strange aliases for nested files

### ISSUE THREAD

**Issue by jacobgardner** [aspect-build/rules_swc#192]
### What happened?

An import that utilizes swc paths (examples/paths/src/modules/moduleA/index.js) is generated with a much longer relative path than should be needed.

Actual:
```javascript
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "moduleA", {
    enumerable: true,
    get: ()=>moduleA
});
const _moduleB = require("../../../../../../../.cache/bazel/_bazel_jacob/0c86d75ab0e68ba3b8d2283871991c92/sandbox/linux-sandbox/66/execroot/aspect_rules_swc/src/modules/moduleB");
const moduleA = ()=>{
    console.log("This is module A");
    (0, _moduleB.moduleB)();
};
```

Expected:
```javascript
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "moduleA", {
    enumerable: true,
    get: ()=>moduleA
});
const _moduleB = require("../moduleB");
const moduleA = ()=>{
    console.log("This is module A");
    (0, _moduleB.moduleB)();
};
```

### Version

Development (host) and target OS/architectures:

Output of `bazel --version`: aspect 5.2.1

Version of the Aspect rules, or other relevant rules from your
`WORKSPACE` or `MODULE.bazel` file:
These are the versions inside of our own project which showed this issue, but the reproducible example is just whatever is in this repo. 
rules_js = 1.23.2
rules_ts = 1.3.3
rules_swc = 1.0.0

Language(s) and/or frameworks involved:
typescript

### How to reproduce

```shell
bazel build //examples/paths:compile
cat bazel-bin/examples/paths/src/modules/moduleA/index.js
```

The root index.js appears to be generated correctly though:

```
   1   │ "use strict";
   2   │ Object.defineProperty(exports, "__esModule", {
   3   │     value: true
   4   │ });
   5   │ const _moduleA = require("./modules/moduleA/index");
   6   │ (0, _moduleA.moduleA)();
```
```


### Any other information?

_No response_

**Comment by jackvincentnz** [comment]
After digging around for a long time it looks like I have encountered the same issue as this one. 

I added a small test for it to show it not working in rules_swc, but looks like we are waiting on the upstream issue.

https://github.com/aspect-build/rules_swc/pull/223

**Comment by mcclayton** [comment]
@jbedard, it appears as though https://github.com/swc-project/swc/issues/4057 has just been fixed and is part of swc release [1.3.98](https://github.com/swc-project/swc/blob/main/CHANGELOG.md#1398---2023-11-21) FYI.

**Comment by alexeagle** [comment]
Another fix was needed in SWC since the original repro I handed them had only one symlink hop while Bazel creates two.

https://github.com/swc-project/swc/pull/8757 just landed, so the next swc release will finally work with Bazel again :)

**Comment by alexeagle** [comment]
Okay we're declaring that this is unfixable in SWC, because we don't have the time or energy to figure out their symlink handling.

HOWEVER there's no benefit to sandboxing SWCCompile actions. They have a single file at a time, and no dependencies, so developers cannot "forget to list some dependencies". With no benefit of better hermeticity therefore, we may as well run these actions outside the sandbox, and in that case this bug goes away since there are no symlinks.

Just add this to your `.bazelrc`:

```
# Workaround SWC issue with symlinks
common --modify_execution_info=SWCCompile=+no-sandbox
```

Leaving this issue open for now until I document this pattern for users to find.

**Comment by alexeagle** [comment]
Long ago, we disabled the sandbox:
https://github.com/aspect-build/rules_swc/blob/29f5a69ec3636920d11c33576f2dc2d295be7a01/.bazelrc#L6-L8
(introduced May 2022 in https://github.com/aspect-build/rules_swc/pull/33)

Our fix for this in #272 is just to restore that behavior. By giving up on sandboxing the SWC action (in particular, in the one-file-input case where hermeticity is not a big problem) then this issue is resolved.

It does introduce a subtle bug where you might not declare the `.swcrc` input at all, yet SWC will locate the config anyway. We consider that a minor issue since you'll discover it's missing when running in a stronger sandbox like a docker or remote spawn.
